test/python/circuit/test_controlled_gate.py

```diff
@@ -768,90 +783,82 @@ def test_mcx_gates(self, num_ctrl_qubits):
         reference = np.zeros(2 ** (num_ctrl_qubits + 1))
         reference[-1] = 1
 
-        for gate in [
-            MCXGrayCode(num_ctrl_qubits),
-            MCXRecursive(num_ctrl_qubits),
-            MCXVChain(num_ctrl_qubits, False),
-            MCXVChain(num_ctrl_qubits, True),
+        for synth_circuit in [
+            synth_mcx_n_dirty_i15(num_ctrl_qubits),
+            synth_mcx_n_clean_m15(num_ctrl_qubits),
+            synth_mcx_1_clean_b95(num_ctrl_qubits),
+            synth_mcx_gray_code(num_ctrl_qubits),
+            synth_mcx_noaux_v24(num_ctrl_qubits),
```

Comment:
Do we need this test? Previously we checked that adding custom MCX variants worked correctly, but now we are checking correctness of synthesis methods, and this does not seem the right place to do that, do you agree?

Reply:
see https://github.com/Qiskit/qiskit/pull/13604/files/6e504de7655f521e0716a43af56e2c96c32f2c0b#r2102580813